### PR TITLE
[APM2-211] Add support for moblie banking OCBC PAO

### PR DESCRIPTION
**Pull request by FhanOmise**
#### 1. Objective

Add support for mobile banking OCBC PAO.

Checkout Page:
Screen Shot 2564-12-02 at 14 44 28

Admin:
image

Screen Shot 2564-12-02 at 10 48 39

**Related information**:
Related task(s): https://omise.atlassian.net/browse/APM2-211

#### 2. Description of change

- Add Mobile Banking payment method.
- Add OCBC PAO Mobile Banking payment type and logo.

#### 3. Quality assurance

Test charge creation for OCBC PAO mobile banking.

#### 4. Impact of the change

Admin will see new payment method 'mobile banking'. If it is selected, checkout page will show mobile banking payment with OCBC PAO in the list.

#### 5. Priority of change

Normal.

#### 6. Additional Notes

### CODE REVIEW

File: includes/backends/class-omise-backend-mobile-banking.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Note: The calculations in this class depend on the countries that
+ *       the available installment payments are based on.
```
Comment (kan98) [inline includes/backends/class-omise-backend-mobile-banking.php:4]: I think you can remove this

File: includes/backends/class-omise-backend-mobile-banking.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ *
+ * @since 3.4
```
Comment (kan98) [inline includes/backends/class-omise-backend-mobile-banking.php:6]: please update to what would be the new version. `4.16` I imagine

File: includes/backends/class-omise-backend-mobile-banking.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+			'mobile_banking_ocbc_pao' => array(
+				'title'              => __( 'OCBC Pay Anyone', 'omise' ),
+				'logo'				 => 'ocbc_pao',
+			),
+		);
+	}
+
+	/**
+	 *
+	 * @return array  of an available installment providers
```
Comment (kan98) [inline includes/backends/class-omise-backend-mobile-banking.php:33]:
```suggestion
	 * @return array  of an available mobile banking providers
```

File: assets/css/omise-css.css [hunk 4]
```diff
@@ -234,6 +234,14 @@ ul.omise-banks-list {
 	border: 4px solid #fff;
 }
 
+.omise-banks-list .mobile-banking-logo {
```
Comment (kan98) [inline assets/css/omise-css.css:237]: on the screenshots you attached, the logo and label look slightly off from the checkbox. Is this something we can fix? Or maybe is this the same as installments too?

File: includes/gateway/class-omise-payment-mobilebanking.php [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+		$this->id                 = 'omise_mobilebanking';
+		$this->has_fields         = true;
+		$this->method_title       = __( 'Omise Mobile Banking', 'omise' );
+		$this->method_description = wp_kses(
+			__( 'Accept payment through <strong>Mobile Banking</strong> via Omise payment gateway.', 'omise' ),
+			array(
+				'strong' => array()
+			)
+		);
```
Comment (kan98) [inline includes/gateway/class-omise-payment-mobilebanking.php:16]: May be useful to add here that specific mobile banking payment methods will be fetched automatically and only show what the merchant has access to. I'm not really sure how best to phrase it though 😅

**Review by tookaa2 — APPROVED** [review]

**Review by kan98 — APPROVED** [review]

**Review by nimitomise — APPROVED** [review]

**Pull request merged**